ui/components/multichain/connected-site-popover/connected-site-popover.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { forwardRef, useContext, RefObject } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  AvatarNetwork,
+  AvatarNetworkSize,
+  Box,
+  ButtonLink,
+  ButtonSecondary,
+  IconName,
+  Popover,
+  Text,
+} from '../../component-library';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import { I18nContext } from '../../../contexts/i18n';
+import { getCurrentNetwork, getOriginOfCurrentTab } from '../../../selectors';
+import { getURLHost } from '../../../helpers/utils/util';
+import { getImageForChainId } from '../../../selectors/multichain';
+import { toggleNetworkMenu } from '../../../store/actions';
+
+type ConnectedSitePopoverProps = {
+  isOpen: boolean;
+  isConnected: boolean;
+  onClick: () => void;
+  onClose: () => void;
+  referenceElement?: RefObject<HTMLElement>;
+};
+
+export const ConnectedSitePopover = forwardRef<
+  HTMLDivElement,
+  ConnectedSitePopoverProps
+>(({ isOpen, isConnected, onClick, onClose, referenceElement }) => {
+  const t = useContext(I18nContext);
+  const activeTabOrigin = useSelector(getOriginOfCurrentTab);
+  const siteName = getURLHost(activeTabOrigin);
+  // TODO: Replace it with networkClient Selector
+  // const activeDomain = useSelector(getAllDomains);
+  // const networkClientId = activeDomain?.[activeTabOrigin];
+  const currentNetwork = useSelector(getCurrentNetwork);
```

Comment:
Do you think we should use the network selectors from `shared/modules/selectors/networks.ts`?

Reply:
Which network Selector are you referring to? I am planning to use the selector when I merge this [PR](https://github.com/MetaMask/metamask-extension/pull/27710)

Reply:
There is kind of mess around network selectors, I see we have the ones from `shared/modules/selectors/networks.ts` and `ui/selectors/selector.js` that are relevant for the network-controller and EVM network data. I'm thinking if we should deprecate one set.

The selector you can also use here is `getProviderConfig`

Reply:
Yes, the end goal is to use getProviderConfig but once I merge the linked PR, I will use it to rely on networkConfigID